modules/tensor_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -81,6 +81,12 @@ addCrackFrontDefinitionParams(InputParameters & params)
                              q_function_type,
                              "The method used to define the integration domain. Options are: " +
                                  q_function_type.getRawNames());
+  params.addParam<UserObjectName>(
+      "crack_front_points_privider",
+      "The UserObject provides the crack front points from XFEM GeometricCutObject");
+  params.addParam<unsigned int>(
+      "number_crack_front_points",
+      "The number of crack front points, only needed if crack_front_points_provider is used.");
```

Comment:
It seems like this might be better in the input for the CrackFrontPointProvider. It would clean up the code on the fracture integral side a bit. I guess it would need to be optional there, and you'd have to generate an error if the points were requested and it wasn't supplied.